builtin/providers/aws/resource_aws_cloudfront_distribution.go

```diff
@@ -0,0 +1,595 @@
+				Computed: true,
+			},
+			"domain_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_modified_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"in_progress_validation_batches": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"etag": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			// retain_on_delete is a non-API attribute that may help facilitate speedy
+			// deletion of a resoruce. It's mainly here for testing purposes, so
+			// enable at your own risk.
+			"retain_on_delete": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudFrontDistributionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+	params := &cloudfront.CreateDistributionInput{
+		DistributionConfig: expandDistributionConfig(d),
+	}
+
+	resp, err := conn.CreateDistribution(params)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.Distribution.Id)
+	return resourceAwsCloudFrontDistributionRead(d, meta)
+}
+
+func resourceAwsCloudFrontDistributionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+	params := &cloudfront.GetDistributionInput{
+		Id: aws.String(d.Id()),
+	}
+
+	resp, err := conn.GetDistribution(params)
+	if err != nil {
+		return err
+	}
+
+	// Update attributes from DistributionConfig
+	flattenDistributionConfig(d, resp.Distribution.DistributionConfig)
+	// Update other attributes outside of DistributionConfig
+	d.SetId(*resp.Distribution.Id)
+	d.Set("active_trusted_signers", flattenActiveTrustedSigners(resp.Distribution.ActiveTrustedSigners))
+	d.Set("status", *resp.Distribution.Status)
+	d.Set("domain_name", *resp.Distribution.DomainName)
+	d.Set("last_modified_time", aws.String(resp.Distribution.LastModifiedTime.String()))
+	d.Set("in_progress_validation_batches", *resp.Distribution.InProgressInvalidationBatches)
+	d.Set("etag", *resp.ETag)
+	return nil
+}
+
+func resourceAwsCloudFrontDistributionUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+	params := &cloudfront.UpdateDistributionInput{
+		Id:                 aws.String(d.Id()),
+		DistributionConfig: expandDistributionConfig(d),
+		IfMatch:            aws.String(d.Get("etag").(string)),
+	}
+	_, err := conn.UpdateDistribution(params)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsCloudFrontDistributionRead(d, meta)
+}
+
+func resourceAwsCloudFrontDistributionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	// manually disable the distribution first
+	d.Set("enabled", false)
+	err := resourceAwsCloudFrontDistributionUpdate(d, meta)
+	if err != nil {
+		return err
+	}
+
+	// skip delete if retain_on_delete is enabled
+	if d.Get("retain_on_delete").(bool) {
```

Comment:
Hey @catsby, yeah it might be something we want to remove from the acceptance tests then, lest the test account get cluttered with disabled distributions. I can add this to the work I'm doing on the rest of the code review stuff if you want.